From 886e44516caa9d26576a6485444a67adc9bef11f Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 10 Dec 2020 22:18:37 +0530
Subject: [PATCH 0953/1921] octeontx2-pf: Add missing mutex lock

mutex_lock is missing in otx2_cgx_features_get
function. Hence add it.

Change-Id: Iebda6e2a8820a73a12e36182e24ecf00c56723d3
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42263
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index b9485839cfd5..3cbd43a10a85 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1142,6 +1142,7 @@ int otx2_cgx_features_get(struct otx2_nic *pfvf)
 	struct msg_req *msg;
 	int err;
 
+	mutex_lock(&pfvf->mbox.lock);
 	msg = otx2_mbox_alloc_msg_cgx_features_get(&pfvf->mbox);
 
 	if (!msg) {
-- 
2.31.1

